src/db.c

```diff
@@ -220,7 +220,7 @@ void dbOverwrite(redisDb *db, robj *key, robj *val) {
     dictSetVal(db->dict, de, val);
 
     if (server.lazyfree_lazy_server_del) {
-        freeObjAsync(old);
+        freeObjAsync(key,old);
```

Comment:
i now realize that we're missing a call to moduleNotifyKeyUnlink here (better be done before calling `dictSetVal`).

which makes me wonder what else are we missing?